flows/webhook.go

```diff
@@ -150,34 +161,36 @@ func newWebhookCallFromResponse(requestTrace string, r *http.Response) (*Webhook
 		w.status = WebhookStatusResponseError
 	}
 
-	// figure out if our Response is something that looks like text from our headers
-	isText := false
-	contentType := r.Header.Get("Content-Type")
-	if strings.Contains(contentType, "text") ||
-		strings.Contains(contentType, "json") ||
-		strings.Contains(contentType, "utf") ||
-		strings.Contains(contentType, "javascript") ||
-		strings.Contains(contentType, "xml") {
-
-		isText = true
-	}
-
-	// only dump the whole body if this looks like text
-	response, err := httputil.DumpResponse(r, isText)
+	// save response dump without body which will be saved separately
+	responseDump, err := httputil.DumpResponse(response, false)
 	if err != nil {
-		return w, err
+		return nil, err
 	}
-	w.response = string(response)
+	w.response = string(responseDump)
+
+	// only save response body's if we have a supported content-type
+	contentType := response.Header.Get("Content-Type")
+	mediaType, _, _ := mime.ParseMediaType(contentType)
+	saveBody := saveResponseContentTypes[mediaType]
+
+	if saveBody {
+		// only read up to our max body bytes limit
+		bodyReader := io.LimitReader(response.Body, int64(maxBodyBytes)+1)
```

Comment:
so even if a webhook returns a 10GB video - we only read up to a certain number of bytes